### PR TITLE
HIVE-21283 Synonyms for the existing functions

### PR DESCRIPTION
**Pull request by rmsmani**
mid for substr
position for Locate

@pvary  Kindly review and merge to master

### CODE REVIEW

**Comment by rmsmani** [comment]
HI @pvary 
Kindly review and merge to master

**Comment by rmsmani** [comment]
Hi @pvary
Please review and merge the code

**Comment by pvary** [comment]
> Hi @pvary
> Please review and merge the code

Sorry, but my schedule is tight nowdays. You might be better off trying to find someone else to review your code than waiting for me to have time to review :(
Sorry :(

**Comment by rmsmani** [comment]
Hi
@jcamachor, @vgarg
Can you please review the code and merge it

**Comment by rmsmani** [comment]
Hi @sankarh
Can you please review the code and merge it

File: ql/src/test/results/clientpositive/udf_substring.q.out [hunk 1]
```diff
@@ -8,7 +8,7 @@ PREHOOK: type: DESCFUNCTION
 POSTHOOK: query: DESCRIBE FUNCTION EXTENDED substring
 POSTHOOK: type: DESCFUNCTION
 substring(str, pos[, len]) - returns the substring of str that starts at pos and is of length len orsubstring(bin, pos[, len]) - returns the slice of byte array that starts at pos and is of length len
-Synonyms: substr
+Synonyms: mid, substr
```
Comment (sankarh) [inline ql/src/test/results/clientpositive/udf_substring.q.out]: Add unit tests for mid and position UDFs.

**Comment by rmsmani** [comment]
No Teset Failures.
image

**Comment by rmsmani** [comment]
HI @sankarh 
Included Unit Test Cases.
Check and  merge the code.

**Comment by rmsmani** [comment]
Hi @sankarh
Kindly review and merge the code

**Comment by sankarh** [comment]
@rmsmani 
I think, there wasn't a green ptest build yet. The build #16541 you highlighted had a testsuit that is timed-out.
TestWebHCatE2e - did not produce a TEST-*.xml file (likely timed out) (batchId=205)
Also, the latest patch HIVE.21283.08.PATCH had a failure too in build #16548.
org.apache.hadoop.hive.cli.TestNegativeCliDriver.testCliDriver[udaf_invalid_place] (batchId=99)

I know, it is flaky tests that are failing but we need a green build to merge a patch.
Please rebase your changes against master and submit a new patch.

**Comment by rmsmani** [comment]
Hi @sankarh
All the TC pertaining to this patch went successful and the failed ones are the falky one.

Still will rebase and submit the patch.

udaf_invalid_place this test got failed in previous build of other Jira patches as well.

**Comment by rmsmani** [comment]
Hi @sankarh
Submitted patch 9 yday night, failed due to yet another Falky test

**Comment by sankarh** [comment]
@rmsmani 
Yes, that is a problem with Hive ptest. Lot of flaky tests. 
But, it was mandated to have a green build before commit. Please re-submit the same patch and try again.

**Comment by rmsmani** [comment]
Hi @sankarh
At last the unit test came green. Please review and merge the code to master

**Comment by rmsmani** [comment]
Source committed to master,so closing

**Pull request closed without merging**